keras_nlp/models/preprocessor.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from tensorflow import keras
+
+from keras_nlp.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_nlp")
+class Preprocessor(keras.layers.Layer):
+    def __init__(
+        self,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+
+    @property
+    def tokenizer(self):
+        """The tokenizer used to tokenize strings."""
+        return self._tokenizer
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "tokenizer": keras.layers.serialize(self.tokenizer),
+                "sequence_length": self.packer.sequence_length,
+                "truncate": self.packer.truncate,
+            }
+        )
+        return config
+
+    @classmethod
+    def from_config(cls, config):
+        if "tokenizer" in config and isinstance(config["tokenizer"], dict):
+            config["tokenizer"] = keras.layers.deserialize(config["tokenizer"])
+        return cls(**config)
+
+    @classproperty
+    def tokenizer_cls(cls):
+        return None
+
+    @classproperty
+    def presets(cls):
+        return {}
+
+    @classmethod
+    def from_preset(
+        cls,
+        preset,
+        sequence_length=None,
+        **kwargs,
+    ):
+        """Instantiate {{model_name}} preprocessor from preset architecture.
+
+        Args:
+            preset: string. Must be one of "{{preset_names}}".
+            sequence_length: int, optional. The length of the packed inputs.
```

Comment:
Don't list this here, but do show the override example below. This will fit with how we document things like `preprocessor` on the task models.

Reply:
The `None` behavior is pretty non-intuitive without an explanation, we definitely need something. I am OK with @mattdangerw 's proposal however.

Reply:
I think I see this as similar to `preprocessor` on classifier's `from_preset`. We do not list `preprocessor` in kwargs or as an argument, yet it's value in `from_preset` is very different than the default in the constructor.

Being overall consistent here seems helpful. We push the mental model of taking advantage of kwargs to override params. Though in all cases good documentation of this seems like a good thing.